src/api/routes/action.js

```diff
@@ -0,0 +1,78 @@
+const express = require("express");
+const router = express.Router();
+
+const checkAuth = require("../middleware/check-auth");
+const checkAdmin = require("../middleware/check-admin");
+const checkPermissions = require("../middleware/check-permissions");
+const logSearch = require("../middleware/log-search");
+
+const ActionController = require("../controllers/ActionController");
+const ActionUserController = require("../controllers/ActionUserController");
+
+/* CRUD */
+
+router.post("/create", checkAuth, ActionController.createAction);
+
+router.patch(
+  "/:action_id",
+  checkAuth,
+  checkPermissions,
+  ActionController.updateAction
+);
+
+router.delete(
+  "/:action_id",
+  checkAuth,
+  checkPermissions,
+  ActionController.cancelAction
+);
+
+/* Admin */
+
+router.post(
+  "/:action_id/approve",
+  checkAuth,
+  checkAdmin,
+  ActionController.approveAction
+);
+
+router.post(
+  "/:action_id/decline",
+  checkAuth,
+  checkAdmin,
+  ActionController.declineAction
+);
+
+/* Attendants */
+
+router.post(
+  "/:action_id/attendees/:user_id",
```

Comment:
This, along with the rest of the endpoints below, should be reformatted to something like `/:action_id/attend`, since we don't really need to specify the user_id as it's extractable by the auth token. I've updated the [project page](https://github.com/din0s/ActionHeroes/projects/1) to reflect this as well.